build/yamls/flow-visibility/flow-visibility.yaml

```diff
@@ -0,0 +1,418 @@
+  name: flow-visibility
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    app: flow-visibility
+  name: flow-visibility
+  namespace: flow-visibility
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  labels:
+    app: flow-visibility
+  name: grafana-role
+  namespace: flow-visibility
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - services
+    verbs:
+      - get
+      - list
+      - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  labels:
+    app: flow-visibility
+  name: grafana-role-binding
+  namespace: flow-visibility
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: grafana-role
+subjects:
+  - kind: ServiceAccount
+    name: grafana
+---
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: grafana-storage
+provisioner: kubernetes.io/no-provisioner
+volumeBindingMode: WaitForFirstConsumer
+reclaimPolicy: Delete
+allowVolumeExpansion: True
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: grafana-pvc
+  namespace: flow-visibility
+spec:
+  storageClassName: grafana-storage
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+---
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: grafana-pv
+  namespace: flow-visibility
+spec:
+  storageClassName: grafana-storage
+  capacity:
+    storage: 2Gi
+  accessModes:
+    - ReadWriteOnce
+  hostPath:
+    path: "/data/grafana"
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: clickhouse-secret
+  namespace: flow-visibility
+type: Opaque
+stringData:
+  username: clickhouse_operator
+  password: clickhouse_operator_password
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: grafana-datasource
+  namespace: flow-visibility
+data:
+  datasource.yaml: |-
+    apiVersion: 1
+    datasources:
+      - name: ClickHouse
+        type: vertamedia-clickhouse-datasource
+        access: proxy
+        url: http://clickhouse-clickhouse.flow-visibility.svc:8123
+        editable: true
+        jsonData:
+          server: clickhouse-clickhouse.flow-visibility.svc
+          port: 9000
+          username: $CH_USERNAME
+        secureJsonData:
+          password: $CH_PASSWORD
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: grafana
+  name: grafana
+  namespace: flow-visibility
+spec:
+  selector:
+    matchLabels:
+      app: grafana
+  template:
+    metadata:
+      labels:
+        app: grafana
+    spec:
+      securityContext:
+        fsGroup: 472
+        supplementalGroups:
+          - 0
+      containers:
+        - name: grafana
+          image: grafana/grafana:8.3.3
+          imagePullPolicy: IfNotPresent
+          env:
+            - name: GF_INSTALL_PLUGINS
+              value: "https://github.com/antrea-io/antrea/releases/download/v1.6.0/grafana-sankey-plugin.zip;antreaflowvisibility-grafana-sankey-plugin,grafana-clickhouse-datasource"
```

Comment:
So, we upload the plugins for 1.6.0 release, even before 1.6.0 is out? Is it a common practice with our release downloads?

Reply:
The plugin will be uploaded when release tag is created. see [here](https://github.com/antrea-io/antrea/blob/af490d89ec1eed6a4b74d252eaa3bab5a91b4ad1/.github/workflows/build_grafana_plugin.yml#L4-L6)

Reply:
If we changed the plugin, will it be auto uploaded?

Reply:
It will not unless we create a new release. But our assumption is that the sankey panel plugin will not have much feature change once it is in use since it will always take input and visualize in a fixed way. 
However, it is indeed a concern if we will have bug-fix in main branch. In that case, if we want to auto upload the plugin whenever it gets changed, we have nowhere to serve it. Do you have any suggestions?

Reply:
I believe there are workflows written by Antonin to check changes and upload, but not sure that applies to code in the repo. @antoninbas should comment.

Reply:
So there is a workflow to upload the plugin zip file as a release asset for each Antrea release, but there is no automated way to update this yaml file when there is a new release? That's a bit weird, because that means someone needs to remember to update this file manually?

Maybe for now it would make more sense to upload new plugins to the antrea.io S3 storage bucket manually when you want to publish a new version, unless it needs to be coupled with Antrea releases. If you do want to couple it to Antrea releases and keep the workflow you have in this PR, it feels like the visibility YAML manifest should also be automatically uploaded as a release asset.

If the plugin was in a different repo, it would be either to version it differently.

Reply:
Out of curiostty, would it be possible to repackage a grafana image which also includes our plugin and publish it like the other antrea images?

Reply:
> Out of curiostty, would it be possible to repackage a grafana image which also includes our plugin and publish it like the other antrea images?

Thanks Salvatore. I haven't manually tried it. I believe that should work. As for a local Grafana, the installed plugins will be placed in the default plugins directory: /var/lib/grafana/plugins. It should have the same effect if we copy the .zip into the directory manually. 

The only concern I have is, if we want to update the Grafana base image version, repackaging an image costs more than changing the version tag in the YAML. Also considering the fact that sankey-plugin won't update frequently, while Grafana updates very frequently. 

I suppose we can use antrea S3 for now, if we don't have major concerns with it. Repackaging the image is definitely something we can try as an alternative if we want to revisit this decision later.